service/lib/agama/http/clients/software.rb

```diff
@@ -77,19 +76,18 @@ def get_resolvables(unique_id, type, optional)
           JSON.parse(get("software/resolvables/#{unique_id}?type=#{type}&optional=#{optional}"))
         end
 
-        def provisions_selected?(_provisions)
-          # TODO: implement it, not sure how it should look like
-          []
+        def provisions_selected?(provisions)
+          provisions.select do |prov|
+            package_installed?(prov)
```

Comment:
Also, this makes one HTTP request per one member of _provisions_. It may be fine for our use cases but we may want a reminder to allow for many provisions?

```suggestion
            # TODO: add bulk API
            package_installed?(prov)
```